homeassistant/components/comfoconnect/config_flow.py

```diff
@@ -0,0 +1,158 @@
+"""Config flow for ComfoConnect integration."""
+import logging
+import secrets
+
+from pycomfoconnect import Bridge, PyComfoConnectNotAllowed
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PIN, CONF_SENSORS, CONF_TOKEN
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import STEP_ID_INIT, STEP_ID_USER
+import homeassistant.helpers.config_validation as cv
+
+from . import ComfoConnectBridge
+from .const import (
+    CONF_USER_AGENT,
+    DEFAULT_NAME,
+    DEFAULT_PIN,
+    DEFAULT_TOKEN,
+    DEFAULT_USER_AGENT,
+    DOMAIN,
+)
+from .sensor import SENSOR_TYPES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass, input_data) -> str:
+    """Validate the user input by trying to connect to the ComfoConnect bridge."""
+    host = input_data[CONF_HOST]
+    bridges = await hass.async_add_executor_job(Bridge.discover, host)
+    if not bridges:
+        raise ConnectionError(f"No bridges found at host {host}")
+    ccb = ComfoConnectBridge(
+        hass,
+        bridge=bridges[0],
+        name=input_data.get(CONF_NAME, DEFAULT_NAME),
+        token=input_data.get(CONF_TOKEN, DEFAULT_TOKEN),
+        friendly_name=input_data.get(CONF_USER_AGENT, DEFAULT_USER_AGENT),
+        pin=input_data.get(CONF_PIN, DEFAULT_PIN),
```

Comment:
These defaults should be the result of the config validation schema, and should not be needed here.